src/transactions/TransactionFrame.cpp

```diff
@@ -644,14 +644,10 @@ TransactionFrame::insertKeysForTxApply(
 void
 TransactionFrame::markResultFailed()
 {
-    // changing "code" causes the xdr struct to be deleted/re-created
-    // As we want to preserve the results, we save them inside a temp object
-    // Also, note that because we're using move operators
-    // mOperations are still valid (they have pointers to the individual
-    // results elements)
-    xdr::xvector<OperationResult> t(std::move(getResult().result.results()));
+    // Changing "code" normally causes the XDR structure to be destructed, then
+    // a different XDR structure is constructed. However, txFAILED and txSUCCESS
+    // have the same underlying field number so this does not occur.
```

Comment:
interesting: pretty sure this used to not be the case for some reason (on Windows for sure), so I had to do this nasty thing in there. Generated code looks fine now though.

You can remove the "sanity check" as this is irrelevant now.